### ISSUE TITLE
The edited message doesn't update on LHN in offline mode

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17212]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to web chrome
2. Turn off your wifi
3. Go to any chat and send a message
4. Now edit the message and notice that it doesn't update on LHN

## Expected Result: 
The edited message should be updated on LHN on offline mode
## Actual Result: 
The edited message doesn't update on LHN on offline mode

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.97-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/230796522-141e4d6a-9d6c-4406-bc00-8f30cee80999.mp4

https://user-images.githubusercontent.com/43996225/230796526-c5e24db1-7304-437f-8e79-f350dd52a96b.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681029679583489

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sonialiap (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by sonialiap** [comment]
Reproducible

**Comment by MelvinBot** [comment]
Triggered auto assignment to @francoisl (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by francoisl** [comment]
Editing a message should be an optimistic action, so I don't think it's supposed to be intentional.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01953a5f80d1aa943c

**Comment by MelvinBot** [comment]
Current assignee @sonialiap is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @fedirjh (`External`)

**Comment by dukenv0307** [comment]
> Could you please consider reformatting your proposal to improve its readability ? Specifically this part :



@fedirjh [Updated proposal](https://github.com/Expensify/App/issues/17212#issuecomment-1503890852)

**Comment by kaushiktd** [comment]
### Please re-state the problem that we are trying to solve in this issue.
The edited message doesn't update on LHN in offline mode

### What is the root cause of that problem?
When the user edited an existing message, editReportComment() is called from below mentioned file:
https://github.com/Expensify/App/blob/a177ba383730a01b4cdb5cf3b1d4f90c2d86a157/src/libs/actions/Report.js#L826
This function is setting up optimisticData only for ReportActions, not for Reports fields. So the last message is not updating in the Report field.

### What changes do you think we should make in order to solve the problem?
We need to add `optimisticData` for the Report field so it can be updated for Report Fields. For this please add the below code in mentioned file: 
https://github.com/Expensify/App/blob/a177ba383730a01b4cdb5cf3b1d4f90c2d86a157/src/libs/actions/Report.js#L871

```
const currentTime = DateUtils.getDBTime();
const optimisticReport = {
    lastVisibleActionCreated: currentTime,
    lastMessageText: ReportUtils.formatReportLastMessageText(htmlForNewComment),
    lastActorEmail: currentUserEmail,
    lastReadTime: currentTime,
};
```

and update `optimisticData` value like below:
 ```
const optimisticData = [
        {
            onyxMethod: CONST.ONYX.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
            value: optimisticReport,
        },
        {
            onyxMethod: CONST.ONYX.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
            value: optimisticReportActions,
        },
    ];
```


### Video File:
https://drive.google.com/file/d/1k54k7EluDa8sBIU1C78QmJtUabV6-dht/view?usp=share_link

**Comment by francoisl** [comment]
👍 @dukenv0307 your proposal sounds good, let's go with it.

**Comment by MelvinBot** [comment]
📣 @dukenv0307 You have been assigned to this job by @francoisl!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by dukenv0307** [comment]
Hi @fedirjh @francoisl [The PR](https://github.com/Expensify/App/pull/17324) is ready for review

**Comment by dukenv0307** [comment]
# Regression Test Proposal

**Bug**: The edited message doesn't update on LHN in offline mode

**Proposed Test Steps:**

1. Login with any account
2. Turn off your wifi or go to Setting --> Preferences and turn on Force offline
3. Verify that you're in offline mode
4. Go to any chat and send a message
5. Edit the last message
6. Verify that the edited message is updated on LHN

Do we 👍 or 👎

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/17574)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by dukenv0307** [comment]
@fedirjh I just created [new PR](https://github.com/Expensify/App/pull/17586) to fix regression. Please help to check

**Comment by francoisl** [comment]
Thanks for the fix @dukenv0307. While we're here, can you please update the regression steps [here](https://github.com/Expensify/App/issues/17212#issuecomment-1507849039) to include testing multiline comments and comments with markdown please?

**Comment by dukenv0307** [comment]
@francoisl @fedirjh  [Regression steps](https://github.com/Expensify/App/issues/17212#issuecomment-1507849039) is updated

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.1-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17324
- https://github.com/Expensify/App/pull/17586


If no regressions arise, payment will be issued on 2023-04-26. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@fedirjh] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@fedirjh] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@fedirjh] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@sonialiap] Determine if we should create a regression test for this bug.
- [ ] [@fedirjh] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@sonialiap] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.2-5 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17586


If no regressions arise, payment will be issued on 2023-04-27. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@fedirjh] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@fedirjh] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@fedirjh] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@sonialiap] Determine if we should create a regression test for this bug.
- [ ] [@fedirjh] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@sonialiap] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
@francoisl, @sonialiap, @fedirjh, @dukenv0307 Eep! 4 days overdue now. Issues have feelings too...

**Comment by sonialiap** [comment]
@dukenv0307 offer sent for fixing the issue (+ bonus for merging under 3 days)
@fedirjh offer sent for reviewing (+ bonus for merging under 3 days, - regression = base price for issue)

**Comment by sonialiap** [comment]
@priya-zha is this your [Upwork profile](https://www.upwork.com/freelancers/~011a4fb6999ba89a4a)? Confirming so that I make sure I send the offer to the correct person :)

**Comment by priya-zha** [comment]
@sonialiap This profile is not mine. My profile is : https://www.upwork.com/freelancers/~011e32317dbbd489e0

**Comment by sonialiap** [comment]
@dukenv0307 paid
@fedirjh paid

@priya-zha thanks for the link. Offer sent for reporting issue

**Comment by priya-zha** [comment]
Thanks accepted.

**Comment by sonialiap** [comment]
@priya-zha paid! :)
